gui/wxpython/iclass/digit.py

```diff
@@ -147,13 +150,23 @@ def CopyMap(self, name, tmp=False):
         poMapInfoNew = pointer(Map_info())
 
         if not tmp:
-            open_fn = Vect_open_new
+            if update:
+                open_fn = Vect_open_update
+            else:
+                open_fn = Vect_open_new
         else:
-            open_fn = Vect_open_tmp_new
-
-        is3D = bool(Vect_is_3d(self.poMapInfo))
-        if open_fn(poMapInfoNew, name, is3D) == -1:
-            return -1
+            if update:
+                open_fn = Vect_open_tmp_update
+            else:
+                open_fn = Vect_open_tmp_new
+
+        if update:
+            if open_fn(poMapInfoNew, name, mapset) == -1:
```

Comment:
It seems unnecessary to pass the mapset to the function, it always needs to be current mapset, in fact, even empty string is valid here and seems to work:
```suggestion
            if open_fn(poMapInfoNew, name, "") == -1:
```

Reply:
Otherwise it looks good!

I would regard the frmt warning as a bug, although possibly harmless, but we don't have to solve that here.

Reply:
> I would regard the frmt warning as a bug, although possibly harmless, but we don't have to solve that here.

Yes, allright. But according the [line 88 in the G__open()](https://grass.osgeo.org/programming7/gis_2open_8c_source.html) function it is seems a warning message, it is okay that the return variable fd (which try open file and return file descriptor) has acquired -1 value. And than that [*G_fopen_old() function](https://grass.osgeo.org/programming7/gis_2open_8c_source.html#l00253) return NULL and finally on the [line 290 in Vect__open_old() function](https://grass.osgeo.org/programming7/vector_2Vlib_2open_8c_source.html#l00162) in the condition, decides based on returning value from *G_fopen_old() function whether it is a native format or not. I don't see a bug in this :-).

Reply:
> It seems unnecessary to pass the mapset to the function, it always needs to be current mapset, in fact, even empty string is valid here and seems to work:

Yes, you're right.

Reply:
> > I would regard the frmt warning as a bug, although possibly harmless, but we don't have to solve that here.
> 
> Yes, allright. But according the [line 88 in the G__open()](https://grass.osgeo.org/programming7/gis_2open_8c_source.html) function it is seems a warning message, it is okay that the return variable fd (which try open file and return file descriptor) has acquired -1 value. And than that [*G_fopen_old() function](https://grass.osgeo.org/programming7/gis_2open_8c_source.html#l00253) return NULL and finally on the [line 290 in Vect__open_old() function](https://grass.osgeo.org/programming7/vector_2Vlib_2open_8c_source.html#l00162) in the condition, decides based on returning value from *G_fopen_old() function whether it is a native format or not. I don't see a bug in this :-).

What I mean here is, it is confusing why you get a warning when you import the training areas when you did everything correctly, it should not behave like that. I haven't looked into this, but I wonder why this happens only for temporary vectors?

Reply:
> What I mean here is, it is confusing why you get a warning when you import the training areas when you did everything correctly, it should not behave like that. I haven't looked into this, but I wonder why this happens only for temporary vectors?

Main difference between [Vect_open_tmp_new()](https://grass.osgeo.org/programming7/vector_2Vlib_2open_8c.html#a25da70e62b2d0d4828b2cbdc90b57f33) function (which don't print this warning message)  and [Vect_open_tmp_update() ](https://grass.osgeo.org/programming7/vector_2Vlib_2open_8c.html#a4d80cc4e331b1d7330b94969df915f6d)function is that first function use different way how to detect vector format (internally call [map_format()](https://grass.osgeo.org/programming7/vector_2Vlib_2open_8c_source.html#l01292) function). Second function internally use [G__open()](https://grass.osgeo.org/programming7/gis_2open_8c_source.html#l00052) function.